runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -104,6 +104,26 @@ void J9::RecognizedCallTransformer::process_java_lang_StringUTF16_toBytes(TR::Tr
    treetop->insertAfter(TR::TreeTop::create(comp(), TR::Node::create(node, TR::treetop, 1, newCallNode)));
    }
 
+void J9::RecognizedCallTransformer::process_java_lang_StrictMath_sqrt(TR::TreeTop* treetop, TR::Node* node)
+   {
+   if (!node->getSymbol()->castToResolvedMethodSymbol()->canReplaceWithHWInstr())
+      {
+      TR::Node* dumpNode = node->getChild(0);
+      TR::Node* valueNode = node->getChild(1);
+
+      anchorNode(dumpNode, treetop);
+      prepareToReplaceNode(node);
+
+      TR::Node::recreateWithoutProperties(node, TR::dsqrt, 1, valueNode, getSymRefTab()->findOrCreateNewArraySymbolRef(node->getSymbolReference()->getOwningMethodSymbol(comp())));
+      TR::TransformUtil::removeTree(comp(), treetop);
+      }
+
+   else
+      {
```

Comment:
Extra newline above. Can you explain what this else path is supposed to do? Why are we recreating the call node if the hardware does not support square root?